libc-top-half/musl/src/thread/pthread_mutexattr_setprotocol.c

```diff
@@ -14,7 +16,9 @@ int pthread_mutexattr_setprotocol(pthread_mutexattr_t *a, int protocol)
 		r = check_pi_result;
 		if (r < 0) {
 			volatile int lk = 0;
+#ifdef __wasilibc_unmodified_upstream
 			r = -__syscall(SYS_futex, &lk, FUTEX_LOCK_PI, 0, 0);
+#endif
```

Comment:
Seems like `check_pi_result` and `r` will always just be -1 here, resulting in `-1` being returned instead of an errno.  

Maybe just remove this entire block and just return `ENOTSUP`?